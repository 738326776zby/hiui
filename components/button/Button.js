import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Provider from '../context'

class Button extends Component {
  static propTypes = {
    type: PropTypes.oneOf(['primary', 'line', 'success', 'danger', 'default', 'warning']),
    size: PropTypes.oneOf(['large', 'small', 'default']),
    appearance: PropTypes.oneOf(['link', 'default', 'line']),
    className: PropTypes.string,
    style: PropTypes.object,
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
    href: PropTypes.string,
    target: PropTypes.oneOf(['_self', '_blank', '_parent', '_top'])
  }

  static defaultProps = {
    prefixCls: 'hi-btn',
    type: 'default',
    disabled: false,
    appearance: 'default',
    size: 'default'
  }

  clickCb () {
    if (this.props.onClick) {
      this.props.onClick()
    }
  }

  render () {
    const {
      type,
      disabled,
      className,
      size,
      prefixCls,
      appearance,
      style,
      title,
      href,
      target,
      theme
    } = this.props
    const classes = classNames(
      'theme__' + theme,
      `${prefixCls}`,
      className && `${className}`,
      // For version < 1.1.0
      (type === 'primary' && appearance === 'line')
        ? `${prefixCls}--type_line`
        : `${prefixCls}--type_${type || 'default'}`,
      appearance && `${prefixCls}--appearance_${appearance || 'default'}`,
      size && `${prefixCls}--size_${size}`,
      disabled && `${prefixCls}--disabled`
    )

    const disabledBool = !!disabled
    return (
      href
        ? <a
          className={classes}
          onClick={() => this.clickCb()}
          style={style}
          title={title}
          href={href}
          target={target}
        >
          {this.props.children}
        </a>
        : <button
          className={classes}
          disabled={disabledBool}
          onClick={() => this.clickCb()}
          style={style}
          title={title}
          type='button'
        >
          {this.props.children}
        </button>
    )
  }
}

export default Provider(Button)
