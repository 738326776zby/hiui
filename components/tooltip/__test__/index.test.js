import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Tooltip from '../'
import Button from '../../button'

describe('Tooltip', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(
      <div>
        <Tooltip title="tooltip top" style={{margin: '0 10px'}}><Button type="line">Tooltip Top</Button></Tooltip>
        <Tooltip title="tooltip right" style={{margin: '0 10px'}} placement="right"><Button type="success">Tooltip Right</Button></Tooltip>
        <Tooltip title="tooltip bottom" style={{margin: '0 10px'}} placement="bottom"><Button type="warning">Tooltip Bottom</Button></Tooltip>
        <Tooltip title="tooltip left" style={{margin: '0 10px'}} placement="left"><Button type="danger">Tooltip Left</Button></Tooltip>
      </div>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should show overlay when trigger is hovered', () => {
    const wrapper = mount(
      <Tooltip title="tooltip top" style={{margin: '0 10px'}}><Button type="line">Tooltip Top</Button></Tooltip>
    )
    wrapper.find('.hi-tooltip').simulate('mouseenter')
    expect(document.querySelectorAll('.hi-popper__container')).toHaveLength(1)
    expect(document.querySelectorAll('.hi-popper__container--hide')).toHaveLength(0)
    wrapper.find('.hi-tooltip').simulate('mouseleave')
    expect(document.querySelectorAll('.hi-popper__container--hide')).toHaveLength(1)
  })
})
