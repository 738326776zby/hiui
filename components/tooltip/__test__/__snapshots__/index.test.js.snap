// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders correctly 1`] = `
<div>
  <div
    className="hi-tooltip"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "margin": "0 10px",
      }
    }
  >
    <button
      className="theme__hiui-blue hi-btn hi-btn--appearance--button hi-btn--size--normal hi-btn--type--line"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Tooltip Top
    </button>
  </div>
  <div
    className="hi-tooltip"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "margin": "0 10px",
      }
    }
  >
    <button
      className="theme__hiui-blue hi-btn hi-btn--appearance--button hi-btn--size--normal hi-btn--type--success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Tooltip Right
    </button>
  </div>
  <div
    className="hi-tooltip"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "margin": "0 10px",
      }
    }
  >
    <button
      className="theme__hiui-blue hi-btn hi-btn--appearance--button hi-btn--size--normal hi-btn--type--warning"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Tooltip Bottom
    </button>
  </div>
  <div
    className="hi-tooltip"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "margin": "0 10px",
      }
    }
  >
    <button
      className="theme__hiui-blue hi-btn hi-btn--appearance--button hi-btn--size--normal hi-btn--type--danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Tooltip Left
    </button>
  </div>
</div>
`;
